components/region_cache_memory_engine/src/background.rs

```diff
@@ -754,29 +760,63 @@ impl Filter {
             default_cf_handle,
             write_cf_handle,
             mvcc_key_prefix: vec![],
-            cached_delete_key: None,
+            cached_mvcc_delete_key: None,
+            cached_skiplist_delete_key: None,
             remove_older: false,
             metrics: FilterMetrics::default(),
         }
     }
 
     fn filter(&mut self, key: &Bytes, value: &Bytes) -> Result<(), String> {
         self.metrics.total += 1;
-        let InternalKey { user_key, .. } = decode_key(key);
+        let InternalKey {
+            user_key, v_type, ..
+        } = decode_key(key);
 
         let (mvcc_key_prefix, commit_ts) = split_ts(user_key)?;
         if commit_ts > self.safe_point {
             return Ok(());
         }
 
+        // Just like what rocksdb compaction filter does, we do not handle internal
+        // keys (representing different MVCC versions of the same user key) that have
+        // been marked as tombstones. However, these keys need to be deleted. 
+        // Since they are below the safe point, we can safely delete them directly now.
+        if v_type == ValueType::Deletion {
```

Comment:
The whole 'if v_type == ValueType::Deletion' can be simplified by the code below. It can be put after 'if let Some(ref cache_skiplist_delete_key) = self.cached_skiplist_delete_key' block.

Reply:
This is for case that two consecutive ValueType::Deletion exists.

Reply:
Please update the comments per our discussion offline.